### ISSUE TITLE
cmd/go: cross-compiling package tests emits errors to stderr yet 0 exit code

### ISSUE THREAD

**Issue by dmitshur** [golang/go#62123]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version devel go1.22-6a1182027c Thu Aug 17 21:40:36 2023 +0000 linux/amd64
</pre>

### Does this issue reproduce with the latest release?

Yes, with Go 1.21.0. (See https://github.com/golang/go/issues/61960#issuecomment-1675139082.)

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE=''
GOARCH='amd64'
GOBIN=''
GOCACHE='/home/gopher/.cache/go-build'
GOENV='/home/gopher/.config/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFLAGS=''
GOHOSTARCH='amd64'
GOHOSTOS='linux'
GOINSECURE=''
GOMODCACHE='/home/gopher/go/pkg/mod'
GONOPROXY=''
GONOSUMDB=''
GOOS='linux'
GOPATH='/home/gopher/go'
GOPRIVATE=''
GOPROXY='https://proxy.golang.org,direct'
GOROOT='/home/gopher/gotip'
GOSUMDB='sum.golang.org'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/home/gopher/gotip/pkg/tool/linux_amd64'
GOVCS=''
GOVERSION='devel go1.22-6a1182027c Thu Aug 17 21:40:36 2023 +0000'
GCCGO='gccgo'
GOAMD64='v1'
AR='ar'
CC='gcc'
CXX='g++'
CGO_ENABLED='1'
GOMOD='/tmp/tmp.mTNFgITEiU/go.mod'
GOWORK=''
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
PKG_CONFIG='pkg-config'
GOGCCFLAGS='-fPIC -m64 -pthread -Wl,--no-gc-sections -fmessage-length=0 -ffile-prefix-map=/tmp/go-build4223926392=/tmp/go-build -gno-record-gcc-switches'
</pre></details>

### What did you do?

I used a linux/amd64 host and attempted to cross-compile a simple Go package and its tests to android/amd64, one of Go's ports. Here's a somewhat minified way to reproduce the issue:

```sh
$ cd $(mktemp -d)
$ go mod init example.com/m
$ echo "package p; import "testing"; func Test(t *testing.T) {}" > p_test.go
$ GOOS=android GOARCH=amd64 go test -c -o=/dev/null
```

### What did you expect to see?

Either successful compilation, no output to stderr, and 0 exit code,
or a clearly failed compilation and a non-zero exit code.

### What did you see instead?

Output to stderr that looks like an error, but exit code 0.

```sh
$ GOOS=android GOARCH=amd64 go test -c -o=/dev/null
# example.com/m.test
loadinternal: cannot find runtime/cgo
$ echo $?
0
```

I spotted this in LUCI (#61960), where we try to rely on `go test -c -o=/dev/null` to catch compilation errors on all supported Go ports.

CC @bcmills, @ianlancetaylor.

**Comment by bcmills** [comment]
That output looks like one or both of:
- #46330
- #31544

See also:
- #57961

**Comment by mknyszek** [comment]
@cherrymui says that this is actually a successful (well, probably :)) compilation, just some additional (possibly confusing) output is printed.

**Comment by cherrymui** [comment]
See also #31544. `loadinternal: cannot find runtime/cgo` is just a print in the linker, not an error. The build actually succeeds, and you should get a binary that is (very likely) working.

**Comment by bcmills** [comment]
The fix for #31544 apparently didn't cover the case of `CGO_ENABLED=0` at all. 😅

(This issue is a symptom of exactly #46330.)
